src/runtime_src/core/tools/xbmgmt2/flash/xospiversal.h

```diff
@@ -24,7 +24,7 @@
 class XOSPIVER_Flasher
 {
 public:
-    XOSPIVER_Flasher(unsigned int device_index);
+    XOSPIVER_Flasher(std::shared_ptr<xrt_core::device> dev);

```

Comment:
Passing shared_ptr by value is a no-no unless you are std::move'ing it into some container.
It is preferable to pass raw pointer (dev.get()) unless you are sharing ownership.

Reply:
My changes got overwritten. I will do this along with Stephen's review comments.